pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -55,7 +55,7 @@
  */
 public class RetentionManager extends ControllerPeriodicTask<Void> {
   public static final long OLD_LLC_SEGMENTS_RETENTION_IN_MILLIS = TimeUnit.DAYS.toMillis(5L);
-  private static final RetryPolicy DEFAULT_RETRY_POLICY = RetryPolicies.exponentialBackoffRetryPolicy(5, 1000L, 2.0f);
+  private static final RetryPolicy DEFAULT_RETRY_POLICY = RetryPolicies.randomDelayRetryPolicy(20, 100L, 200L);
```

Comment:
This is quite aggressive. Not sure about its side effect

Reply:
The reason for changing the retry policy is simply to make it more resilient along with serialising the update for idealState during deletion, currently when a lot of segments are being uploaded and the retention manger is trying to delete segments, the the retry is applied 5 times using exponential backoff and it fails. 

For example :

```
2024/05/16 01:59:39.041 ERROR [RetentionManager] [pool-20-thread-4] Failed to clean up the segment lineage. (tableName = tableXXX_OFFLINE)
org.apache.pinot.spi.utils.retry.AttemptsExceededException: Operation failed after 5 attempts
        at org.apache.pinot.spi.utils.retry.BaseRetryPolicy.attempt(BaseRetryPolicy.java:65) ~[startree-pinot-all-1.2.0-ST.10.1-jar-with-dependencies.jar:1.2.0-ST.10.1-8711a0aa760c824774f3ceb1a2913878a31071ec]
        at org.apache.pinot.controller.helix.core.retention.RetentionManager.manageSegmentLineageCleanupForTable(RetentionManager.java:195) ~[startree-pinot-all-1.2.0-ST.10.1-jar-with-dependencies.jar:1.2.0-ST.10.1-8711a0aa760c824774f3ceb1a2913878a31071ec]
        at org.apache.pinot.controller.helix.core.retention.RetentionManager.processTable(RetentionMan
```

Reply:
Okay on further inspection, I do not see much value in adding random delays with the current parameters since the maximum wait time will be `200*20 = 4000ms` where the minimum delay now is `2000 + 4000 + 8000 + 16000 + 32000 = 62000ms` and the update is failing even then. 

I think increasing the number of attempts in exponential delay should be sufficient. What do you think @Jackie-Jiang @snleee ?

Reply:
Main motivation:
- Both retention manager & segment upload were both trying to update idealstate
- Retention Manager was trying to update the idealstate without lock and used a simple exponential back-off
- Segment upload side idealstate update is using `DEFAULT_TABLE_IDEALSTATES_UPDATE_RETRY_POLICY = RetryPolicies.randomDelayRetryPolicy(20, 100L, 200L);`
- Segment upload side were updating idealstate after holding the lock

Observation
- Retention Manager failed a lot of time after 5 attempts while most of idealstate updates via segment upload went through.


I thought about this briefly. I feel that as long as we hold the lock, retry logic may not matter too much. @Jackie-Jiang how do you think?

Reply:
Merging it and we can see how it works. Holding the lock itself might not be enough because there is no guarantee which controller is taking the segment upload